src/pages/[...slug].tsx

```diff
@@ -1,15 +1,40 @@
 import type { NextPage } from 'next';
 import { useRouter } from 'next/router';
+import { useMemo } from 'react';
+import { Switcher } from 'src/components/switcher/Switcher';
+import { Governance } from 'src/features/governance/components/Governance';
 import { Swap } from 'src/features/swap/components/Swap';
+import { Validators } from 'src/features/validators/components/Validators';
+import { CenteredLayout } from 'src/layout/CenteredLayout';
 import { Mode } from 'src/types';
 
 const SwapPage: NextPage = () => {
   const router = useRouter();
   const { slug } = router.query as { slug?: string[] };
   const mode = slug ? slug[0] : Mode.stake;
 
-  if (typeof mode !== 'string' || !Object.values(Mode).includes(mode as Mode)) return null;
-  return <Swap mode={mode as Mode} />;
+  const page = useMemo(() => {
```

Comment:
why the memo?

Reply:
honestly just like better the switch + return syntax better than switch + assign + break, happy to change it if wanted. I wish you could do `const x = switch(y) {... }`